src/pytest_cpp/catch2.py

```diff
@@ -96,24 +96,6 @@ def run_test(
                 )
             except subprocess.CalledProcessError as e:
                 output = e.output
-                if e.returncode != 1:
-                    msg = (
-                        "Internal Error: calling {executable} "
-                        "for test {test_id} failed (returncode={returncode}):\n"
-                        "{output}"
-                    )
-                    failure = Catch2Failure(
-                        executable,
-                        0,
-                        msg.format(
-                            executable=executable,
-                            test_id=test_id,
-                            output=e.output,
-                            returncode=e.returncode,
-                        ),
-                    )
-
-                    return [failure], output
```

Comment:
If you get an exit code of 3 (because 3 tests fail), this would force the parser to load the executable, which would be in binary and cause an error.

```
    return [x.rstrip() for x in f.readlines()[index_above : index + 1]]
                                ^^^^^^^^^^^^^
  File "<frozen codecs>", line 322, in decode
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xe0 in position 24: invalid continuation byte
```

Reply:
Ahh definitely, this seems like an oversight.